core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -164,6 +169,20 @@ public void process(final Archive<?> applicationArchive, final TestClass testCla
         tryToAdd(libs, libraryContainer, "WEB-INF/lib/scala-library-", "cucumber.api.scala.ScalaDsl", "scala.App");
     }
 
+    private boolean archiveContains(Archive<?> archive, Class<?> clazz) {

```

Comment:
that's why you need to try both names (of previous comment) but will be way more efficient on real projects with 10* or 100* of resources - that's the case I had in mind and idea was to not iterate to find but just check if it is there directly (same for war iteration BTW)